### PR TITLE
Price in CRM should be 0 by default if no price entered - hint message

### PR DESCRIPTION
**Pull request by jeremie74**

### CODE REVIEW

File: libs/admin/src/lib/admin-panel/pages/offers/shell/contract-form/contract-form.component.html [hunk 1]
```diff
@@ -10,9 +10,11 @@
     </mat-form-field>
     <mat-form-field appearance="outline">
       <mat-label>Title price</mat-label>
-      <input type="number" matInput formControlName="price" required min="0"/>
+      <input type="number" matInput formControlName="price" />
```
Comment (RemcoSimonides) [inline libs/admin/src/lib/admin-panel/pages/offers/shell/contract-form/contract-form.component.html:13]: Shouldn't the `min="0"` stay? Without it you can enter sub-zero numbers I think

File: libs/admin/src/lib/admin-panel/pages/offers/shell/contract-form/contract-form.component.html [hunk 2]
```diff
@@ -10,9 +10,11 @@
...
       <mat-icon matPrefix *ngIf="currency" [svgIcon]="currency"></mat-icon>
-      <mat-error *ngIf="form.hasError('required', 'price')">Price field is required!</mat-error>
+      <!-- Please live this comment if PM change idea 
+      <mat-error *ngIf="form.hasError('required', 'price')">Price field is required!</mat-error> -->
+      <mat-hint>If no price entered. Seller will recieve active price.</mat-hint>
```
Comment (RemcoSimonides) [inline libs/admin/src/lib/admin-panel/pages/offers/shell/contract-form/contract-form.component.html:17]: comma, receive* and Asking Price*  (receive is incorrectly written in the design - correct way is receive*)
"If no price entered, Seller will receive Asking Price"

File: libs/admin/src/lib/admin-panel/pages/offers/shell/contract-form/contract-form.component.html [hunk 3]
```diff
@@ -10,9 +10,11 @@
     </mat-form-field>
     <mat-form-field appearance="outline">
       <mat-label>Title price</mat-label>
-      <input type="number" matInput formControlName="price" required min="0"/>
+      <input type="number" matInput formControlName="price" />
       <mat-icon matPrefix *ngIf="currency" [svgIcon]="currency"></mat-icon>
-      <mat-error *ngIf="form.hasError('required', 'price')">Price field is required!</mat-error>
+      <!-- Please live this comment if PM change idea 
+      <mat-error *ngIf="form.hasError('required', 'price')">Price field is required!</mat-error> -->
```
Comment (RemcoSimonides) [inline libs/admin/src/lib/admin-panel/pages/offers/shell/contract-form/contract-form.component.html:16]: You can remove it. We aim to keep the repo clean from unused code ;)

**Review by RemcoSimonides — CHANGES_REQUESTED** [review]

File: libs/admin/src/lib/admin-panel/pages/offers/shell/contract-form/contract-form.component.html (hunk 2 above)
Comment (jeremie74) [reply libs/admin/src/lib/admin-panel/pages/offers/shell/contract-form/contract-form.component.html:17]: Ok i copied it from an other page, i had a doubt and yes its receive

File: libs/admin/src/lib/admin-panel/pages/offers/shell/contract-form/contract-form.component.html (hunk 3 above)
Comment (jeremie74) [reply libs/admin/src/lib/admin-panel/pages/offers/shell/contract-form/contract-form.component.html:16]: Yes but is someone change advice, we will have to put it, but ok ill do it

File: libs/admin/src/lib/admin-panel/pages/offers/shell/contract-form/contract-form.component.html (hunk 1 above)
Comment (jeremie74) [reply libs/admin/src/lib/admin-panel/pages/offers/shell/contract-form/contract-form.component.html]: yes it's true

File: libs/admin/src/lib/admin-panel/pages/offers/shell/contract-form/contract-form.component.ts [hunk 4]
```diff
@@ -108,8 +108,13 @@ export class ContractFormComponent implements OnInit {
           await this.contractService.update(contractId, { titleId, termIds  }, { write });
         }
         if(price !== this.income.price) {
-          await this.incomeService.update(contractId, { price }, { write }); // Update the price in the batch
+          if(price === null) {
+            await this.incomeService.update(contractId, { price: 0 }, { write });
+          } else {
+            await this.incomeService.update(contractId, { price }, { write });
+          }
```
Comment (RemcoSimonides) [inline libs/admin/src/lib/admin-panel/pages/offers/shell/contract-form/contract-form.component.ts:115]: After small interaction with Francois we decided to not change the price to `0` and keep `null` instead. Can you undo this again? ^^

**Review by RemcoSimonides — APPROVED** [review]

File: libs/admin/src/lib/admin-panel/pages/offers/shell/contract-form/contract-form.component.ts (hunk 4 above)
Comment (jeremie74) [reply libs/admin/src/lib/admin-panel/pages/offers/shell/contract-form/contract-form.component.ts:115]: I understand it's a strange request

**Pull request merged**